src/measure/OSRunner.cpp

```diff
@@ -784,6 +784,141 @@ namespace measure {
     return result;
   }
 
+  Json::Value OSRunner::getArgumentValues(std::vector<OSArgument>& script_arguments, const std::map<std::string, OSArgument>& user_arguments) {
```

Comment:
You mean in terms of the case/switch statement stuff? Not in terms of the function of the methods, right?

Reply:
https://github.com/NREL/OpenStudio/blob/c66049bce31effb11b90c845a26a4dd5959eb427/src/workflow/ApplyMeasure.cpp#L44-L55

https://github.com/NREL/OpenStudio/blob/c66049bce31effb11b90c845a26a4dd5959eb427/src/workflow/OSWorkflow.cpp#L160-L189

The Variant versus OSArgument stuff is sooooo confusing.

Also, we have OSArgument.toJSON. I feel like there should be a way to avoid sprinkling this type of things all over the place.

Reply:
Hmm so maybe there's a way to use OSArgument::toJSON on each script argument, and then assign its appropriate value to the new `argument_values` hash?

Although, looks like toJSON never assigns `.valueAsXXX`, which comes from `user_arguments`.